scripts/gen_cfb_font_header.py

```diff
@@ -0,0 +1,160 @@
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import sys
+
+from PIL import ImageFont
+from PIL import Image
+from PIL import ImageDraw
+
+def generate_element(image, charcode):
+    """Generate CFB font element for a given character code from an image"""
+    blackwhite = image.convert("1", dither=Image.NONE)
+    pixels = blackwhite.load()
+
+    if args.dump:
+        blackwhite.save("{}_{}.png".format(args.name, charcode))
+
+    args.output.write("""\t/* {:d} */\n\t{{\n""".format(charcode))
+
+    for col in range(0, args.width):
+        args.output.write("\t\t")
+        for octet in range(0, int(args.height / 8)):
+            value = ""
+            for bit in range(0, 8):
+                row = octet * 8 + bit
+                if pixels[col, row]:
+                    value = "0" + value
+                else:
+                    value = "1" + value
+            args.output.write("0x{:02x},".format(int(value, 2)))
+        args.output.write("\n")
+    args.output.write("\t},\n")
+
+def extract_font_glyphs():
+    """Extract font glyphs from a TrueType/OpenType font file"""
+    font = ImageFont.truetype(args.input, args.size)
+    for i in range(args.first, args.last + 1):
+        image = Image.new("RGB", (args.width, args.height), (255, 255, 255))
+        draw = ImageDraw.Draw(image)
+        draw.text((0, 0), chr(i), (0, 0, 0), font=font)
+        generate_element(image, i)
+
+def extract_image_glyphs():
+    """Extract font glyphs from an image file"""
+    image = Image.open(args.input)
+
+    x_offset = 0
+    for i in range(args.first, args.last + 1):
+        glyph = image.crop((x_offset, 0, x_offset + args.width, args.height))
+        generate_element(glyph, i)
+        x_offset += args.width
+
+def generate_header():
+    """Generate CFB font header file"""
+    guard = "__CFB_FONT_{:s}_{:d}{:d}_H__".format(args.name.upper(), args.width,
+                                                  args.height)
+
+    args.output.write("""/*
+ * This file was automatically generated using the following command:
+ * {cmd}
+ *
+ */
+
+#ifndef {guard}
+#define {guard}
+
+#include <zephyr.h>
+#include <display/cfb.h>
+
+const u8_t cfb_font_{name:s}_{width:d}{height:d}[{elem:d}][{b:.0f}] = {{\n"""
+                      .format(cmd=" ".join(sys.argv),
+                              guard=guard,
+                              name=args.name,
+                              width=args.width,
+                              height=args.height,
+                              elem=args.last - args.first + 1,
+                              b=args.width / 8 * args.height))
+
+    if args.type == "font":
+        extract_font_glyphs()
+    elif args.type == "image":
+        extract_image_glyphs()
+    elif args.input.name.lower().endswith((".otf", ".otc", ".ttf", ".ttc")):
+        extract_font_glyphs()
+    else:
+        extract_image_glyphs()
+
+    args.output.write("""
+}};
+
+FONT_ENTRY_DEFINE({name},
```

Comment:
`name` should be uniquely, maybe {name}_{width}{height}?

Reply:
@jfischer-phytec-iot Good catch. Fixed. I must have accidentally changed that during one of the rewrites.